Controller/DemoController.php

```diff
@@ -156,9 +156,17 @@ public function listBlogPostsAction( $locationId )
      */
     public function showBlogPostAction( $locationId, $viewType, $layout = false, array $params = array() )
     {
-        // We need the author, whatever the view type is.
         $repository = $this->getRepository();
-        $location = $repository->getLocationService()->loadLocation( $locationId );
+        // When previewing first draft there is no location id
+        if ( $locationId === null && isset( $params['isPreview'] ) && $params['isPreview'] == true && isset( $params['location'] ) )
```

Comment:
We could use what is at `$params['location']`, if it is a valid location, instead of loading at all.